aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,118 @@
+
+## Package
+To include the middleware in your project, add a reference to the  [`Microsoft.AspNetCore.ResponseCaching`](https://www.nuget.org/packages/Microsoft.AspNetCore.ResponseCaching/) package. The middleware is available for projects that target `.NETFramework 4.5.1` or `.NETStandard 1.3` or higher.
+
+## Extensions
+In `ConfgureServices`, add the middleware to your service collection.
+
+[!code-csharp[Main](middleware/sample/Startup.cs?name=snippet1)]
+
+Configure the application to use the middleware when processing requests. The sample application adds a `Cache-Control` header to the response that will cache cacheable responses for up to 10 seconds. The sample also sends a `Vary` header to configure the cache to serve the response only if the `Accept-Encoding` header of subsequent requests matches that from the original request.
+
+[!code-csharp[Main](middleware/sample/Startup.cs?name=snippet2)]
+
+The position of this middleware relative to other middleware in the pipeline is important. Any terminal middleware placed before this middleware will prevent the Response Caching Middleware from caching or serving the response. For example, if you place [Static File Middleware](xref:fundamentals/static-files) before this middleware, your static files will not be cached by the middleware. If you place Static File Middleware after this middleware, your static files will be cached.
+
+>[!NOTE]
+> If you place Static File Middleware before Response Caching Middleware, you can provide a prefix to the middleware that will force it to ignore requests with a path that doesn't match your prefix. For example, you can prefix your application's static file links with **assets** (`http://www.myapp.com/assets/staticfile.txt`) and configure the middleware to ignore any requests that do not include the **assets** prefix, `app.UseStaticFiles("/assets")`. Note that "**assets**" is only part of the path in your links to the static files. Your application will not have an **assets** folder, and your static files should remain in the **wwwroot** folder of your deployed application (**wwwroot\staticfile.txt**).
+
+> The Response Caching Middleware only caches 200 (OK) server responses. Any other responses, including [error pages](xref:fundamentals/error-handling), will be ignored by the middleware.
+
+## Options
+The middleware offers two options for controlling response caching.
+
+Option | Default Value
+--- | ---
+UseCaseSensitivePaths | <p>Determines if responses will be cached on case-sensitive paths.</p><p>The default value is `false`.</p>
+MaximumBodySize | <p>The largest cacheable size for the response body in bytes.</p>The default value is `64 * 1024 * 1024` [64 MB (67,108,864 bytes)].</p>
+
+The following example configures the middleware to cache responses smaller than or equal to 1 MB (1,024 bytes) using case-sensitive paths, storing the responses to `/page1` and `/Page1` separately.
```

Comment:
Yikes! ... this is why we have **reviews!** :smile:

Let's just clean this up by stating in-text "1,024 bytes" and leave it at that. That matches the code line below. We don't need to get into Kibibytes (KiB) here.

For my future reference, is MS using the IEC naming in docs? cc/ @danroth27